promql/engine.go

```diff
@@ -1048,6 +1094,25 @@ func (ev *evaluator) eval(expr Expr) Value {
 			panic(fmt.Errorf("cannot do range evaluation of matrix selector"))
 		}
 		return ev.matrixSelector(e)
+
+	case *SubqueryExpr:
+		newEv := &evaluator{
+			startTimestamp: ev.endTimestamp - int64(e.Range/time.Millisecond),
```

Comment:
startTimestamp?

This should also be aligned with the step

Reply:
If I am understanding this correctly, `startTimestamp = endTimestamp - (some_int * interval)`? 

If so, should `startTimestamp` be before or after `(end - Range)`.

Reply:
Remember that this is a range query evaluation, so you need data from before the startTimestamp.

In addition the startTimestamp should be aligned with the step of the subquery, so the query/query_range invairant isn't broken.

Reply:
I have made some changes to align it with the step. Is it right, or am I still missing something?